File: backend/src/main/java/com/example/Flicktionary/domain/post/dto/PostResponseDto.java
package com.example.Flicktionary.domain.post.dto;

import com.example.Flicktionary.domain.post.entity.Post;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
@Builder
public class PostResponseDto {
    private Long id;
    private Long userAccountId;
    private String nickname;
    private String title;
    private String content;
    private LocalDateTime createdAt;
    private Boolean isSpoiler;

    public static PostResponseDto fromEntity(Post post) {

        // 유저가 탈퇴할 경우 "탈퇴한 회원"으로 처리
        String nickname = (post.getUserAccount() != null) ? post.getUserAccount().getNickname() : "탈퇴한 회원";

        return PostResponseDto.builder()
                .id(post.getId())
                // 유저가 탈퇴할 경우 null처리
                .userAccountId(post.getUserAccount() != null ? post.getUserAccount().getId() : null)
                .title(post.getTitle())
                .content(post.getContent())
                .createdAt(post.getCreatedAt())
                .isSpoiler(post.getIsSpoiler())
                .nickname(nickname)
                .build();
    }
}

File: backend/src/main/kotlin/com/example/Flicktionary/domain/post/dto/PostResponseDto.kt
package com.example.Flicktionary.domain.post.dto

import com.example.Flicktionary.domain.post.entity.Post
import com.fasterxml.jackson.annotation.JsonProperty
import java.time.LocalDateTime

data class PostResponseDto(
    val id: Long? = null,
    val userAccountId: Long? = null,
    val nickname: String,
    val title: String,
    val content: String,
    val createdAt: LocalDateTime? = null,
    @JsonProperty("isSpoiler")
    val isSpoiler: Boolean? = null
) {
    companion object {
        fun fromEntity(post: Post): PostResponseDto {
            // 유저가 탈퇴할 경우 "탈퇴한 회원"으로 처리
            val nickname = post.userAccount?.nickname ?: "탈퇴한 회원"
            val userAccountId = post.userAccount?.id

            return PostResponseDto(
                id = post.id,
                userAccountId = userAccountId,
                title = post.title,
                content = post.content,
                createdAt = post.createdAt,
                isSpoiler = post.isSpoiler,
                nickname = nickname
            )
        }
    }
}
